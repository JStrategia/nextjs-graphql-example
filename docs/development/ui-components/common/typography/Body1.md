# Body1

Renders a `span` tag with theme style and given text content.

## Usage

```tsx
function MyComponent() {
  return <Body1>Some text</Body1>;
}
```
