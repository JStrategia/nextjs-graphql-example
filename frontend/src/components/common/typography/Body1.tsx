import styled from 'styled-components';
import Typography from 'components/common/typography/Typography';

const Body1 = styled.span`
  ${Typography};
`;

export default Body1;
