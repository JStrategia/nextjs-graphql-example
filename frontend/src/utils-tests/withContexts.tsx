import React from 'react';
import ThemeProvider from 'contexts/ThemeProvider';

function withContexts(children: React.ReactNode) {
  return <ThemeProvider>{children}</ThemeProvider>;
}

export default withContexts;
