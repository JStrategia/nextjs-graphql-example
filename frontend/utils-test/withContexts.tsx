import React, { ReactNode } from 'react';
import ThemeProvider from '../contexts/ThemeProvider';

function withContexts(children: ReactNode) {
  return <ThemeProvider>{children}</ThemeProvider>;
}

export default withContexts;
