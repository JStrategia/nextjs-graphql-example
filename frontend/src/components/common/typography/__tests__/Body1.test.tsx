import React from 'react';
import { create } from 'react-test-renderer';
import Body1 from 'components/common/typography/Body1';
import withContexts from 'utils-tests/withContexts';

describe('Body1', () => {
  describe('Given no text', () => {
    describe('When rendering', () => {
      const render = create(withContexts(<Body1 />)).toJSON();

      test('Then match snapshot', () => {
        expect(render).toMatchSnapshot();
      });
    });
  });

  describe('Given text', () => {
    const text = 'some text';

    describe('When rendering', () => {
      const render = create(withContexts(<Body1>{text}</Body1>)).toJSON();

      test('Then match snapshot', () => {
        expect(render).toMatchSnapshot();
      });
    });
  });
});
