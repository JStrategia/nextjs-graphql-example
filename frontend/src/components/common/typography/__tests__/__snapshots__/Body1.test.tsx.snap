// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body1 Given no text When rendering Then match snapshot 1`] = `
<span
  className="sc-bczRLJ dqVkrV"
/>
`;

exports[`Body1 Given text When rendering Then match snapshot 1`] = `
<span
  className="sc-bczRLJ dqVkrV"
>
  some text
</span>
`;
